docs/upgrade-to-3.0.md

```diff
@@ -0,0 +1,25 @@
+# Upgrading to 3.0
+
```

Comment:
Add a callout here saying that 3.0 is an upcoming release, and that this is a work-in-progress page and not released yet.